### PR TITLE
Improve handling of `test_split_mode='none'` and `val_split_mode='none'`

### PR DESCRIPTION
**Pull request by djdameln**
# Description

Fix for another problem that came up during manual testing of the datamodules branch. The `none` option for `test_split_mode` and `val_split_mode` was not handled properly.

- Prevents sampling normal images from train set to test set when the default test set does not contain normals and `test_split_mode` is `none`.
- Ensures that `limit_val_batches` is set to 0.0 when `val_split_mode` is `none`, to prevent the trainer from trying to run validation on a non-existing validation set.
- `from_test` and `same_as_test` not allowed for `val_split_mode` when test_split_mode is `none` to prevent sampling from non-existing test set.
- Using `test_split_mode` to determine if trainer should test after training.

## Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks
